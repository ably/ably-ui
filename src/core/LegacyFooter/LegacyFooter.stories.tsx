import { delay, http, HttpResponse } from "msw";

import LegacyFooter from "./LegacyFooter";

import ablyStack from "../images/ably-stack.svg";
import bestSupport from "../images/g2-best-support-2025.svg";
import highPerformer from "../images/g2-high-performer-2025.svg";
import usersMostLikely from "../images/g2-users-most-likely-to-recommend-2025.svg";
import bestMeetsRequirements from "../images/g2-best-meets-requirements-2025.svg";

const statusUrl = "https://ntqy1wz94gjv.statuspage.io/api/v2/status.json";

export default {
  title: "Components/LegacyFooter",
  component: LegacyFooter,
  parameters: {
    layout: "fullscreen",
    msw: {
      handlers: {
        status: http.get(statusUrl, async () => {
          await delay();

          return HttpResponse.json({
            status: {
              indicator: "none",
            },
          });
        }),
      },
    },
  },
  args: {
    paths: {
      ablyStack,
      bestSupport,
      highPerformer,
      usersMostLikely,
      bestMeetsRequirements,
    },
    statusUrl: statusUrl,
  },
};

export const Default = {};
