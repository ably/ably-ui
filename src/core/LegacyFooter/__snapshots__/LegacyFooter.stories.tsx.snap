// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/LegacyFooter Default smoke-test 1`] = `
<footer class="bg-light-grey font-sans antialiased leading-normal"
        data-id="footer"
>
  <div class="max-w-screen-xl mx-auto py-32 sm:py-40 md:py-64 ui-grid-gap ui-grid-px grid grid-cols-6">
    <div class="col-span-full md:col-span-2">
      <div class="flex flex-row p-menu-row-snug">
        <img class="mr-24 -mt-16"
             src="data:image/svg+xml,%3csvg%20width='49'%20height='43'%20viewBox='0%200%2049%2043'%20fill='none'%20xmlns='http://www.w3.org/2000/svg'%3e%3cg%20id='platform'%3e%3cpath%20id='grid'%20fill-rule='evenodd'%20clip-rule='evenodd'%20d='M34.6654%2023.0921L24.5365%2017.7703L0.841309%2030.2199L24.5365%2042.8736L48.2317%2030.2199L35.0096%2023.2729L34.8557%2022.9869L34.6654%2023.0921ZM32.5886%2024.24L24.5365%2020.0094L15.947%2024.5224L24.2312%2028.8595L32.5886%2024.24ZM13.814%2025.6431L5.07579%2030.2342L13.5536%2034.7615L22.1523%2030.0086L13.814%2025.6431ZM15.6371%2035.8742L24.5365%2040.6266L33.3962%2035.8953L24.2562%2031.1101L15.6371%2035.8742ZM35.5119%2034.7655L43.9972%2030.2342L34.6888%2025.3435L26.335%2029.961L35.5119%2034.7655Z'%20fill='url(%23paint0_linear)'/%3e%3cpath%20id='flat'%20d='M0.99707%2024.2554L24.5366%2011.9324L48.0762%2024.2554L24.5366%2036.7805L0.99707%2024.2554Z'%20fill='url(%23paint0_linear)'%20fill-opacity='0.75'/%3e%3cpath%20id='solid'%20fill-rule='evenodd'%20clip-rule='evenodd'%20d='M24.5365%205.84314L48.2317%2018.2927L24.5365%2030.9464L0.841309%2018.2927L24.5365%205.84314ZM5.07579%2018.3071L24.5365%2028.6995L43.9972%2018.3071L24.5365%208.08229L5.07579%2018.3071Z'%20fill='%234B3B35'/%3e%3cpath%20id='dashed'%20fill-rule='evenodd'%20clip-rule='evenodd'%20d='M25.1461%200.873535L26.4966%201.58312L25.5937%203.34787L25.1461%203.11268L24.6985%203.34787L23.7956%201.58312L25.1461%200.873535ZM22.9003%204.29263L21.1022%205.2374L20.1992%203.47265L21.9974%202.52788L22.9003%204.29263ZM29.19%205.2374L27.3919%204.29263L28.2948%202.52788L30.093%203.47265L29.19%205.2374ZM32.7864%207.12693L30.9882%206.18216L31.8911%204.41742L33.6893%205.36218L32.7864%207.12693ZM19.304%206.18217L17.5058%207.12693L16.6029%205.36218L18.4011%204.41742L19.304%206.18217ZM36.3827%209.01646L34.5845%208.0717L35.4874%206.30695L37.2856%207.25171L36.3827%209.01646ZM15.7077%208.0717L13.9095%209.01646L13.0066%207.25171L14.8048%206.30695L15.7077%208.0717ZM12.1114%209.96123L10.3132%2010.906L9.41027%209.14125L11.2084%208.19648L12.1114%209.96123ZM39.979%2010.906L38.1808%209.96123L39.0838%208.19648L40.8819%209.14125L39.979%2010.906ZM8.51503%2011.8508L6.71687%2012.7955L5.81395%2011.0308L7.61211%2010.086L8.51503%2011.8508ZM43.5753%2012.7955L41.7772%2011.8508L42.6801%2010.086L44.4782%2011.0308L43.5753%2012.7955ZM4.34688%2012.6227L4.01579%2011.9755L1.45093%2013.3231L4.00999%2014.6897L4.34772%2014.0403L4.91871%2013.7403L4.7129%2013.338L4.92451%2012.9311L4.34688%2012.6227ZM45.9445%2014.0403L45.3735%2013.7403L45.5793%2013.338L45.3677%2012.9311L45.9453%2012.6227L46.2764%2011.9755L48.8413%2013.3231L46.2822%2014.6897L45.9445%2014.0403ZM41.7714%2014.8516L43.5695%2013.8914L44.484%2015.65L42.6859%2016.6102L41.7714%2014.8516ZM6.72267%2013.8914L8.52083%2014.8516L7.60631%2016.6102L5.80815%2015.65L6.72267%2013.8914ZM38.175%2016.7721L39.9732%2015.8119L40.8877%2017.5705L39.0896%2018.5307L38.175%2016.7721ZM10.319%2015.8119L12.1172%2016.7721L11.2026%2018.5307L9.40447%2017.5705L10.319%2015.8119ZM34.5787%2018.6926L36.3769%2017.7324L37.2914%2019.491L35.4932%2020.4513L34.5787%2018.6926ZM13.9153%2017.7324L15.7135%2018.6927L14.799%2020.4513L13.0008%2019.491L13.9153%2017.7324ZM17.5116%2019.6529L19.3098%2020.6132L18.3953%2022.3718L16.5971%2021.4115L17.5116%2019.6529ZM30.9824%2020.6132L32.7806%2019.6529L33.6951%2021.4115L31.8969%2022.3718L30.9824%2020.6132ZM21.108%2021.5734L22.9061%2022.5337L21.9916%2024.2923L20.1934%2023.332L21.108%2021.5734ZM27.3861%2022.5337L29.1842%2021.5734L30.0988%2023.332L28.3006%2024.2923L27.3861%2022.5337ZM25.1461%2023.7299L24.7043%2023.4939L23.7898%2025.2525L25.1461%2025.9768L26.5024%2025.2525L25.5879%2023.4939L25.1461%2023.7299Z'%20fill='%23ADA6A3'/%3e%3c/g%3e%3cdefs%3e%3clinearGradient%20id='paint0_linear'%20x1='35.3254'%20y1='25.3504'%20x2='14.0934'%20y2='28.7416'%20gradientUnits='userSpaceOnUse'%3e%3cstop%20stop-color='%23FF0000'/%3e%3cstop%20offset='1'%20stop-color='%23FB5519'/%3e%3c/linearGradient%3e%3c/defs%3e%3c/svg%3e"
             alt="Ably homepage"
        >
        <h2 class="ui-text-overline2 col-span-full">
          The Ably Platform
        </h2>
      </div>
      <div class="md:col-span-4 md:w-3/4 xs:w-3/5 w-full">
        <p class="ui-text-p3 font-bold py-16 p-menu-row-snug">
          Easily power any realtime experience in your application via a simple API that handles everything realtime.
        </p>
      </div>
      <ul class="grid grid-cols-1">
        <li class="p-menu-row-snug">
          <a href="/pubsub"
             class="ui-footer-menu-row-link"
          >
            Pub/sub messaging
          </a>
        </li>
        <li class="p-menu-row-snug">
          <a href="/push-notifications"
             class="ui-footer-menu-row-link"
          >
            Push notifications
          </a>
        </li>
        <li class="p-menu-row-snug">
          <a href="/integrations"
             class="ui-footer-menu-row-link"
          >
            Third-party integrations
          </a>
        </li>
        <li class="p-menu-row-snug">
          <a href="/protocols"
             class="ui-footer-menu-row-link"
          >
            Multiple protocol messaging
          </a>
        </li>
      </ul>
    </div>
    <div class="col-span-full xs:col-span-3 md:col-span-1">
      <h2 class="ui-footer-col-title">
        Ably is for
      </h2>
      <ul>
        <li class="p-menu-row-snug">
          <a href="/solutions/asset-tracking"
             class="ui-footer-menu-row-link"
          >
            Ably Asset Tracking
          </a>
        </li>
        <li class="p-menu-row-snug">
          <a href="/solutions/extend-kafka-to-the-edge"
             class="ui-footer-menu-row-link"
          >
            Extend Kafka to the edge
          </a>
        </li>
        <li class="p-menu-row-snug">
          <a href="/solutions/edtech"
             class="ui-footer-menu-row-link"
          >
            EdTech
          </a>
        </li>
        <li class="p-menu-row-snug">
          <a href="/solutions/automotive-logistics-and-mobility"
             class="ui-footer-menu-row-link"
          >
            Automotive, Logistics, &amp; Mobility
          </a>
        </li>
        <li class="p-menu-row-snug">
          <a href="/solutions/b2b-platforms"
             class="ui-footer-menu-row-link"
          >
            B2B Platforms
          </a>
        </li>
        <li class="p-menu-row-snug">
          <a href="/solutions/healthcare"
             class="ui-footer-menu-row-link"
          >
            Healthcare
          </a>
        </li>
        <li class="p-menu-row-snug">
          <a href="/solutions/ecommerce-and-retail"
             class="ui-footer-menu-row-link"
          >
            eCommerce &amp; Retail
          </a>
        </li>
        <li class="p-menu-row-snug">
          <a href="/solutions/sports-and-media"
             class="ui-footer-menu-row-link"
          >
            Sports &amp; Media
          </a>
        </li>
        <li class="p-menu-row-snug">
          <a href="/solutions/gaming"
             class="ui-footer-menu-row-link"
          >
            Gaming
          </a>
        </li>
        <li class="p-menu-row-snug">
          <a href="/solutions/iot-and-connected-devices"
             class="ui-footer-menu-row-link"
          >
            IoT &amp; Connected Devices
          </a>
        </li>
      </ul>
    </div>
    <div class="col-span-full xs:col-span-3 md:col-span-1">
      <h2 class="ui-footer-col-title">
        Developers
      </h2>
      <ul>
        <li class="p-menu-row-snug">
          <a href="/docs/quick-start-guide"
             class="ui-footer-menu-row-link"
          >
            Start in 5 minutes
          </a>
        </li>
        <li class="p-menu-row-snug">
          <a href="/docs/"
             class="ui-footer-menu-row-link"
          >
            Documentation
          </a>
        </li>
        <li class="p-menu-row-snug">
          <a href="/tutorials"
             class="ui-footer-menu-row-link"
          >
            Tutorials
          </a>
        </li>
        <li class="p-menu-row-snug">
          <a href="https://changelog.ably.com/"
             class="ui-footer-menu-row-link"
          >
            Changelog
          </a>
        </li>
        <li class="p-menu-row-snug">
          <a href="/support"
             class="ui-footer-menu-row-link"
          >
            Support &amp; FAQs
          </a>
        </li>
        <li class="p-menu-row-snug">
          <a href="/docs/sdks"
             class="ui-footer-menu-row-link"
          >
            SDKs
          </a>
        </li>
        <li class="p-menu-row-snug">
          <a class="ui-footer-menu-row-link flex items-center gap-4"
             href="https://status.ably.com/"
             target="_blank"
             rel="noreferrer"
          >
            System status
            <span class="inline-flex h-8 aspect-square rounded-full bg-neutral-500 animate-pulse">
            </span>
          </a>
        </li>
      </ul>
    </div>
    <div class="col-span-full xs:col-span-3 md:col-span-1">
      <h2 class="ui-footer-col-title">
        WHY ABLY
      </h2>
      <ul>
        <li class="p-menu-row-snug">
          <a href="/customers"
             class="ui-footer-menu-row-link"
          >
            Customers
          </a>
        </li>
        <li class="p-menu-row-snug">
          <a href="/case-studies"
             class="ui-footer-menu-row-link"
          >
            Case Studies
          </a>
        </li>
        <li class="p-menu-row-snug">
          <a href="/four-pillars-of-dependability"
             class="ui-footer-menu-row-link"
          >
            Four Pillars of Dependability
          </a>
        </li>
        <li class="p-menu-row-snug">
          <a href="/compare"
             class="ui-footer-menu-row-link"
          >
            Compare our tech
          </a>
        </li>
        <li class="p-menu-row-snug">
          <a href="/protocols"
             class="ui-footer-menu-row-link"
          >
            Multi protocol support
          </a>
        </li>
        <li class="p-menu-row-snug">
          <a href="/integrations"
             class="ui-footer-menu-row-link"
          >
            Third-party integrations
          </a>
        </li>
      </ul>
    </div>
    <div class="col-span-full xs:col-span-3 md:col-span-1">
      <h2 class="ui-footer-col-title">
        ABOUT
      </h2>
      <ul>
        <li class="p-menu-row-snug">
          <a href="/about"
             class="ui-footer-menu-row-link"
          >
            About Ably
          </a>
        </li>
        <li class="p-menu-row-snug">
          <a href="/pricing"
             class="ui-footer-menu-row-link"
          >
            Pricing
          </a>
        </li>
        <li class="p-menu-row-snug">
          <a href="/blog"
             class="ui-footer-menu-row-link"
          >
            Blog
          </a>
        </li>
        <li class="p-menu-row-snug">
          <a href="/careers"
             class="ui-footer-menu-row-link"
          >
            Careers
          </a>
        </li>
        <li class="p-menu-row-snug">
          <a href="/open-policy"
             class="ui-footer-menu-row-link"
          >
            Open protocol policy
          </a>
        </li>
        <li class="p-menu-row-snug">
          <a href="/press-center"
             class="ui-footer-menu-row-link"
          >
            Press &amp; Media
          </a>
        </li>
        <li class="p-menu-row-snug">
          <a href="/contact"
             class="ui-footer-menu-row-link"
          >
            Contact us
          </a>
        </li>
      </ul>
    </div>
  </div>
  <div class="max-w-screen-xl ui-grid-px mx-auto">
    <hr class="border-t border-mid-grey my-0">
  </div>
  <div class="max-w-screen-xl mx-auto py-16 grid ui-grid-gap ui-grid-px sm:grid-cols-2">
    <div class="md:flex md:items-center">
      <div class="flex flex-col md:flex-row flex-auto ml-8 sm:col-span-1 md:col-span-2">
        <div class>
          <div class="flex pb-24">
            <a class="h-24 pr-24 text-cool-black hover:text-icon-twitter"
               href="https://twitter.com/ablyrealtime"
               title="Ably on X"
            >
              <svg class
                   style="width: 1.5rem; height: 1.5rem;"
              >
                <use xlink:href="#sprite-icon-social-x">
                </use>
              </svg>
            </a>
            <a class="h-24 pr-24 text-cool-black hover:text-icon-linkedin"
               href="https://www.linkedin.com/company/ably-realtime"
               title="Ably on LinkedIn"
            >
              <svg class
                   style="width: 1.5rem; height: 1.5rem;"
              >
                <use xlink:href="#sprite-icon-social-linkedin">
                </use>
              </svg>
            </a>
            <a class="h-24 pr-24 text-cool-black hover:text-icon-github"
               href="https://github.com/ably/"
               title="Ably on Github"
            >
              <svg class
                   style="width: 1.5rem; height: 1.5rem;"
              >
                <use xlink:href="#sprite-icon-social-github">
                </use>
              </svg>
            </a>
            <a class="h-24 pr-24 text-cool-black hover:text-icon-discord"
               href="https://discord.gg/jwBPhEZ9g5"
               title="Ably on Discord"
            >
              <svg class
                   style="width: 1.5rem; height: 1.5rem;"
              >
                <use xlink:href="#sprite-icon-social-discord">
                </use>
              </svg>
            </a>
          </div>
        </div>
        <div class="xs:hidden sm:block ui-footer-glassdoor">
          <div class="flex sm:pt-24 md:pt-0 sm:border-t sm:border-l-0 md:border-t-0 md:border-l sm:border-mid-grey sm:w-3/4 md:w-full md:pl-24">
            <a href="https://www.glassdoor.co.uk/Overview/Working-at-Ably-EI_IE2184188.11,15.htm"
               class="h-24 text-cool-black hover:text-icon-glassdoor"
               title="Ably reviews on glassdoor"
            >
              <svg class
                   style="width: 1.5rem; height: 1.5rem;"
              >
                <use xlink:href="#sprite-icon-social-glassdoor">
                </use>
              </svg>
            </a>
            <div class="pl-16 text-menu3 font-sans font-medium">
              <strong class="block font-medium">
                We're hiring!
              </strong>
              <a href="https://www.glassdoor.co.uk/Overview/Working-at-Ably-EI_IE2184188.11,15.htm"
                 class="ui-footer-link"
              >
                Learn more at Glassdoor
              </a>
            </div>
          </div>
        </div>
      </div>
      <div class="xs:block sm:hidden">
        <div class="border-t border-mid-grey w-full">
        </div>
        <div class="flex py-24">
          <a class="h-24 pr-16 text-cool-black hover:text-icon-glassdoor"
             href="https://www.glassdoor.co.uk/Overview/Working-at-Ably-EI_IE2184188.11,15.htm"
             title="Ably reviews on glassdoor"
          >
            <svg class
                 style="width: 1.5rem; height: 1.5rem;"
            >
              <use xlink:href="#sprite-icon-social-glassdoor">
              </use>
            </svg>
          </a>
          <div class="text-menu3 font-sans font-medium">
            <strong class="block font-medium">
              We're hiring!
            </strong>
            <a href="https://www.glassdoor.co.uk/Overview/Working-at-Ably-EI_IE2184188.11,15.htm"
               class="ui-footer-link"
            >
              Learn more at Glassdoor
            </a>
          </div>
        </div>
      </div>
    </div>
    <div class="col-span-full sm:col-span-1 inline-flex sm:ml-auto sm:items-center">
      <img class="mr-24 h-80"
           src="http://127.0.0.1:6007/assets/g2-best-support-2025-BmlhmO3C.svg"
           alt="Best Support 2025"
      >
      <img class="mr-24 h-80"
           src="http://127.0.0.1:6007/assets/g2-high-performer-2025-C8Ft0OCd.svg"
           alt="High Performer 2025"
      >
      <img class="mr-24 h-80"
           src="http://127.0.0.1:6007/assets/g2-users-most-likely-to-recommend-2025-BQfu7dbr.svg"
           alt="Users Most Likely To Recommend 2025"
      >
      <img class="mr-24 h-80"
           src="http://127.0.0.1:6007/assets/g2-best-meets-requirements-2025-BShFIyI6.svg"
           alt="Best Meets Requirements 2025"
      >
    </div>
  </div>
  <div class="max-w-screen-xl ui-grid-px mx-auto">
    <hr class="border-t border-mid-grey my-0">
  </div>
  <div class="max-w-screen-xl mx-auto py-24 sm:py-40 md:py-32 md:grid md:grid-cols-2 ui-grid-gap ui-grid-px">
    <div class="flex ml-8 col-span-full sm:col-span-1 md:pb-16 items-center ui-footer-bottom-links">
      <div class="flex">
        <a href="/privacy"
           class="pr-24 ui-footer-link"
        >
          Cookies
        </a>
        <a href="/legals"
           class="pr-24 ui-footer-link"
        >
          Legals
        </a>
        <a href="/data-protection"
           class="pr-24 ui-footer-link"
        >
          Data Protection
        </a>
        <a href="/privacy"
           class="ui-footer-link"
        >
          Privacy
        </a>
      </div>
    </div>
    <div class="xs:grid xs:grid-cols-2 sm:grid-cols-4 xs:pl-16 sm:pl-8 md:justify-items-end">
      <div class="flex mr-24">
        <svg class="text-active-orange bg-white rounded-full mr-12 ui-footer-tick-icon"
             style="width: 1.5rem; height: 1.5rem;"
        >
          <use xlink:href="#sprite-icon-gui-check-outline">
          </use>
        </svg>
        <div>
          <p class="ui-footer-compliance-text font-bold whitespace-nowrap">
            SOC 2 Type 2
          </p>
          <p class="ui-footer-compliance-text font-medium mb-24">
            Certified
          </p>
        </div>
      </div>
      <div class="flex mr-24 md:col-start-2">
        <svg class="text-active-orange bg-white rounded-full mr-12 ui-footer-tick-icon"
             style="width: 1.5rem; height: 1.5rem;"
        >
          <use xlink:href="#sprite-icon-gui-check-outline">
          </use>
        </svg>
        <div>
          <p class="ui-footer-compliance-text font-bold whitespace-nowrap">
            HIPAA
          </p>
          <p class="ui-footer-compliance-text font-medium mb-24">
            Compliant
          </p>
        </div>
      </div>
      <div class="flex mr-24 md:col-start-3">
        <svg class="text-active-orange bg-white rounded-full mr-12 ui-footer-tick-icon"
             style="width: 1.5rem; height: 1.5rem;"
        >
          <use xlink:href="#sprite-icon-gui-check-outline">
          </use>
        </svg>
        <div>
          <p class="ui-footer-compliance-text font-bold whitespace-nowrap">
            EU GDPR
          </p>
          <p class="ui-footer-compliance-text font-medium mb-24">
            Certified
          </p>
        </div>
      </div>
      <div class="flex mr-24 md:col-start-4">
        <svg class="text-active-orange bg-white rounded-full mr-12 ui-footer-tick-icon"
             style="width: 1.5rem; height: 1.5rem;"
        >
          <use xlink:href="#sprite-icon-gui-check-outline">
          </use>
        </svg>
        <div>
          <p class="ui-footer-compliance-text font-bold whitespace-nowrap">
            256-bit AES
          </p>
          <p class="ui-footer-compliance-text font-medium mb-24">
            Encryption
          </p>
        </div>
      </div>
    </div>
  </div>
</footer>
`;
